Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```
@@ -0,0 +1,459 @@
+
+    return filtered_events
+
+
+def prepare_raw_res_and_load_json(raw_res: str) -> dict:
+    raw_res = raw_res.replace("}\n{", ",")
+    if not raw_res.startswith("["):
+        raw_res = f"[{raw_res}]"
+    return json.loads(raw_res)
+
+
+def get_events_command(client: Client, integration_context: dict):
+    events: list[dict] = []
+    next_fetch: dict[str, str] = integration_context.get("next_fetch", {})
+
+    try:
+        raw_res = client.get_events(payload=next_fetch)
+        json_res = prepare_raw_res_and_load_json(raw_res)
+    except DemistoException as e:
+        if e.res is not None and e.res.status_code == 401:
+            demisto.info(
+                "Unauthorized access token, trying to obtain a new access token"
+            )
+            raise UnauthorizedToken
+        elif e.res is not None and e.res.status_code == 410:
+            raise NextPointingNotAvailable
+        raise e
+
+    for chunk in json_res:
+        events.extend(chunk["events"])
+    next_hash = json_res[0].get("next", "") if json_res else ""
+
+    if not events:
+        demisto.info("Not events returned")
+        return
+
+    demisto.info(f"Starting event filtering. Initial number of events: {len(events)}")
+    filtered_events = filter_duplicate_events(events, integration_context)
+    demisto.info(f"Filtering completed. Total number of events: {len(filtered_events)}")
+
+    filtered_events.extend(integration_context.get("last_fetch_events", []))
+
+    try:
+        push_events(filtered_events)
+        demisto.info("pushing passed successfully")
+    except Exception as e:
+        update_new_integration_context(
+            filtered_events=filtered_events,
+            next_hash=next_hash,
+            include_last_fetch_events=True,
+            last_integration_context=integration_context,
+        )
+        demisto.info(
+            f"Failed to push events to XSIAM, The integration_context updated. Error: {e}"
+        )
+        raise e
```

`raise DemistoException("here be message") from e` instead of logging